docs/framework/windows-services/walkthrough-creating-a-windows-service-application-in-the-component-designer.md

```diff
@@ -309,37 +309,88 @@ This article demonstrates how to create a simple Windows Service application in
-    ```  
-  
-     This code sets the event source and log name according to the supplied startup parameters, or uses default values if no arguments are supplied.  
-  
-3.  To specify the command-line arguments, add the following code to the `ProjectInstaller` class in ProjectInstaller.cs or ProjectInstaller.vb:  
-  
-    ```csharp  
-    protected override void OnBeforeInstall(IDictionary savedState)        {            string parameter = "MySource1\" \"MyLogFile1";            Context.Parameters["assemblypath"] = "\"" + Context.Parameters["assemblypath"] + "\" \"" + parameter + "\"";            base.OnBeforeInstall(savedState);        }  
-    ```  
-  
-    ```vb  
-    Protected Overrides Sub OnBeforeInstall(ByVal savedState As IDictionary)        Dim parameter As String = "MySource1"" ""MyLogFile1"        Context.Parameters("assemblypath") = """" + Context.Parameters("assemblypath") + """ """ + parameter + """"        MyBase.OnBeforeInstall(savedState)    End Sub  
-    ```  
-  
-     This code modifies the **ImagePath** registry key, which typically contains the full path to the executable for the Windows Service, by adding the default parameter values. The quotation marks around the path (and around each individual parameter) are required for the service to start up correctly. To change the startup parameters for this Windows Service, users can change the parameters given in the **ImagePath** registry key, although the better way is to change it programmatically and expose the functionality to users in a friendly way (for example, in a management or configuration utility).  
+```csharp  
+public MyNewService(string[] args)
+{
+    InitializeComponent();
+    string eventSourceName = "MySource";
+    string logName = "MyNewLog";
+    if (args.Count() > 0) 
+    {
+        eventSourceName = args[0];
+    }
+    if (args.Count() > 1)
+    {
+        logName = args[1];
+    }
+    eventLog1 = new System.Diagnostics.EventLog();
+    if (!System.Diagnostics.EventLog.SourceExists(eventSourceName))
+    {
+        System.Diagnostics.EventLog.CreateEventSource(eventSourceName, logName);
+    }
+    eventLog1.Source = eventSourceName;
+    eventLog1.Log = logName;        
+}
+```  
+  
+```vb  
+Public Sub New(ByVal cmdArgs() As String)
+    InitializeComponent()
+    Dim eventSourceName As String = "MySource"
+    Dim logName As String = "MyNewLog"
+    If (cmdArgs.Count() > 0) Then
+        eventSourceName = cmdArgs(0)
+    End If
+    If (cmdArgs.Count() > 1) Then
+        logName = cmdArgs(1)
+    End If
+    eventLog1 = New System.Diagnostics.EventLog()
+    If (Not System.Diagnostics.EventLog.SourceExists(eventSourceName)) Then
+        System.Diagnostics.EventLog.CreateEventSource(eventSourceName, logName)
```

Comment:
I don't see this @rpetrusha. It looks the same as others. Or am I missing something?
![image](https://user-images.githubusercontent.com/12971179/27935866-9a399b60-6262-11e7-86f8-d84bc82db4e4.png)